simple-payment-example-revamp

bdeklerk0 opened the pull request:
- General
  ~~- Added `node-server` and `php-server` scripts to `package.json` so that each server can be started with `npm run node-server` or `npm run php-server` respectively.~~
  - Added `dotenv` to `package.json`.

- index.html
  - Revised styling - Removed Bootstrap. 
  - Added Tailwind CSS with CDN. 
    - Added custom utility classes for with `tailwind.config.js`. 
  - Improved responsiveness. 
  - Created `./assets` directory for CSS, JS, and images. 
  - Moved `<style>` CSS code section to `/assets/styles.css` 
  - Refactored and optimized `<script>` JS code section 
    - Created functions to reduce code duplication as well as improve efficiency


- Adding PHP related projects to the **[Monorepo](https://github.com/gettilled/tilled-example-monorepo)**
	- ~~- PHP Laravel Server~~
	  ~~- Created a new Laravel project, located in `/tilled-php-laravel`~~
	  ~~- Created `TilledService.php` to handle the Tilled API credentials for requests~~
	  ~~- Created `PaymentController.php` which creates the paymentIntent, and sends the `client_secret` back to the client~~
	  ~~- Added routes for `/` and `/secret/{id}` in `tilled-php-laravel/routes/api.php` and `tilled-php-laravel/routes/web.php`~~
	  ~~- Added `tilled` alias to `tilled-php-laravel/config/app.php`, so that the values can be accessed via `config('tilled')`~~
	    ~~- It uses `env('TILLED_SECRET_KEY')` for the `secret_api_key` value and `'https://sandbox-api.tilled.com'` for the `base_url` value~~
	  ~~- Index page is located at `tilled-php-laravel/resources/views/index.blade.php` ~~
		~~- Same functionality as `index.html` but with Blade templating~~
	      ~~- `accountId` and `publishableKey` variables are defined in the `.env` file, and are accessed using Laravel's `env()` helper function~~
	    ~~- CSS, JS, and images are located in `tilled-php-laravel/public/assets`~~
			~~- CSS, JS, and images are called using Blade's `asset()` helper function~~

- README.md
  - Updated README.md with more clear instructions on getting configuration values and testing payments ~~for running the Node.js and PHP Laravel servers.~~
  - Added instructions for creating a `.env` for both Node.js ~~and PHP Laravel servers.~~
  - Updated images in README.md to reflect new styling.
  - Added new hyperlinks to README.md for `# What's Next?` section.

AllenWoerner commented:
@bdeklerk0 where will this live at ultimately?

dpatton1992 commented:
I still need to test it, but everything looks good!

AllenWoerner commented:
> > @bdeklerk0 where will this live at ultimately?
> 
> @AllenWoerner In this repo. It will replace the existing SPE. The initial Node.js server still exists, but we have added the PHP Laravel server.

@dpatton1992 should we think about taking the opp to move this to the other repo and redirect it? Do we need to have Tyler/Engineering look this over as well? I totally trust yalls judgment but being this is the last remaining project that has engineering ties I just wanted to double check.

dpatton1992 commented on tilled-php-laravel/resources/views/index.blade.php:
```diff
@@ -0,0 +1,590 @@
+    <!-- Included here for simplicity of example -->
+    <script>
+      // Replace with your own account ID and publishable key, these should be defined in your .env file
+      const accountId = "{{ env('TILLED_ACCOUNT_ID') }}"; || "acct_...";
```
A wild ; appeard 😄 
```
const accountId = "{{ env('TILLED_ACCOUNT_ID') }}" || "acct_...";
```
fixes it

bdeklerk0 replied:
Removed it now. Thank you!

bdeklerk0 commented:
@AllenWoerner @dpatton1992 I have removed the PHP server, adjusted the package.json (to remove the scripts), and adjusted the readme.

I will be adding the PHP server as a folder in the mono-repo.

dpatton1992 reviewed (approved):
LGTM
